### PR TITLE
Multiple jobs

### PR DESCRIPTION
**Pull request by betobetodev**
Created because the last merge was incorrectly done.

### CODE REVIEW

**Review by iurygregory — CHANGES_REQUESTED** [review]
Don't need the file .spark_progress.py.swp

File: monitor/plugins/spark_progress.py [hunk 1]
```diff
@@ -30,58 +30,111 @@
...
+
+        application_progress_error = {}
+
+        # Init
+        jobs = job_request.json()
+        jobs.reverse()
+
+        if not len(jobs) == 0:
+            current_job = jobs[self.current_job_id]
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 2]
```diff
@@ -30,58 +30,111 @@
...
+        if not len(jobs) == 0:
+            current_job = jobs[self.current_job_id]
+            
+            if current_job['status'] == 'FAILED':
+                self.current_job_id = len(jobs) - 1
+
+            elif current_job['status'] == 'SUCCEEDED':
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces after the last ")"

File: monitor/plugins/spark_progress.py [hunk 3]
```diff
@@ -30,58 +30,111 @@
...
+
+            elif current_job['status'] == 'RUNNING':
+                # Job Progress
+                job_progress = (current_job['numCompletedTasks']
+                                / float(current_job['numTasks']))
+
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 4]
```diff
@@ -30,58 +30,111 @@
...
+                job_progress = (current_job['numCompletedTasks']
+                                / float(current_job['numTasks']))
+
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+                
+                # Reference Value
+                ref_value = (elapsed_time / self.job_expected_time)
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 5]
```diff
@@ -30,58 +30,111 @@
...
+                # Elapsed Time
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))
+                
+                # Reference Value
+                ref_value = (elapsed_time / self.job_expected_time)
+                
+                # Error
+                error = job_progress - ref_value
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 6]
```diff
@@ -30,58 +30,111 @@
...
+                
+                application_progress_error['name'] = 'application-progress.\
+                                                      error'
+
+                application_progress_error['value'] = error
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+
+                print application_progress_error['value']
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 7]
```diff
@@ -30,58 +30,111 @@
...
+                                                      error'
+
+                application_progress_error['value'] = error
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+
+                print application_progress_error['value']
+                
+                self.monasca.send_metrics([application_progress_error])
+
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: empty spaces

File: monitor/plugins/spark_progress.py [hunk 8]
```diff
@@ -30,58 +30,111 @@
...
-            # The Monasca metric must have the 3 following fields to be created: name, value and timestamp,
-            # but also is possible to increment the metrics identities - using dimensions -  and informations -
-            # using value_meta, a dictionary that contains aditional information about the measurement  .
-            application_progress_error['name'] = 'application-progress.error'
-            application_progress_error['value'] = error
-            application_progress_error['timestamp'] = time.time() * 1000
-            application_progress_error['dimensions'] = self.dimensions
-            # Sending the metric to Monasca
-            self.monasca.send_metrics([application_progress_error])
+        date_time = date_time.replace(tzinfo=None)
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: I didn't get why we need to apply another replace and set tzinfo to None

File: monitor/plugins/spark_progress.py [hunk 9]
```diff
@@ -30,58 +30,111 @@
...
+
+            job_request = requests.get(self.submission_url
+                          + ':4040/api/v1/applications/'
+                          + self.app_id + '/jobs')
+
             self._publish_measurement(job_request)
 
         except Exception as ex:
-            print "Error: No application found for %s. %s remaining attempts" % (self.app_id, self.attempts)
+            # FIXME: @armstrongmsg - self.attempts is not updated anywhere.
```
Comment (iurygregory) [inline monitor/plugins/spark_progress.py]: This fixme should be fixed in the next commits?

**Review by iurygregory — CHANGES_REQUESTED** [review]

File: monitor/plugins/spark_progress.py (hunk 9 above)
Comment (betobetodev) [reply monitor/plugins/spark_progress.py]: Yes.

File: monitor/plugins/spark_progress.py (hunk 1 above)
Comment (betobetodev) [reply monitor/plugins/spark_progress.py]: Fixed.

File: monitor/plugins/spark_progress.py [hunk 10]
```diff
@@ -30,58 +30,108 @@
...
+        try:
+            local_tz = tzlocal.get_localzone()
+
+        except Exception as e:
+            local_tz = "America/Recife"
+            local_tz = pytz.timezone(local_tz)
+
+        date_time = datetime.strptime(gmt_timestamp, '%Y-%m-%dT%H:%M:%S.%fGMT')
+        date_time = date_time.replace(tzinfo=pytz.utc).astimezone(local_tz)
+        date_time = date_time.replace(tzinfo=None)
```
Comment (tellesnobrega) [inline monitor/plugins/spark_progress.py:117]: I agree with Iury, why so many replaces here?

Comment (betobetodev) [reply monitor/plugins/spark_progress.py:117]: The master was working with these replaces. I just reallocated them. I think this is something necessary to adjust the format of received timestamp. Do you think i should insert all the 'replaces' in just one line? @tellesnobrega

Comment (tellesnobrega) [reply monitor/plugins/spark_progress.py:117]: No, this is fine.

**Pull request merged**